kernel: bump 5.10 to 5.10.172

graysky2 opened the pull request:
Removed upstreamed:
        backport-5.10/804-0001-net-Remove-WARN_ON_ONCE-sk-sk_forward_alloc-from-sk_.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.171&id=3e4bbd1f38a8d35bd2d3aaffdb5f6ada546b669a

hauke commented on target/linux/generic/backport-5.10/811-v6.1-0001-nvmem-core-Fix-memleak-in-nvmem_register.patch:
```diff
@@ -1,52 +0,0 @@
-nvmem_validate_keepouts() after device_register() and let
-the device core deal with cleaning name in error cases.
-
-Fixes: de0534df9347 ("nvmem: core: fix error handling while validating keepout regions")
```
I think this patch is still needed, it references this patch: target/linux/generic/backport-5.10/805-v5.15-0003-nvmem-core-fix-error-handling-while-validating-keepo.patch

graysky2 replied:
Thanks, @hauke.  I took a stab at modifying it.  Please see the latest push.

graysky2 committed:
kernel: bump 5.10 to 5.10.169

Manually rebased:
        backport-5.10/811-v6.1-0001-nvmem-core-Fix-memleak-in-nvmem_register.patch

Removed upstreamed:
        backport-5.10/811-v6.1-0003-nvmem-core-add-error-handling-for-dev_set_name.patch[1]
	patches-5.10/070-net-bgmac-fix-BCM5358-support-by-setting-correct-fla.patch[2]

Add fix:
	target/linux/generic/backport-5.10/804-0001-net-Remove-WARN_ON_ONCE-sk-sk_forward_alloc-from-sk_.patch[3]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.169&id=a19a0f67dbb89ad2bfc466f2003841acba645884
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.169&id=a5c51e0c3202820192db3f3809e072f3ca2b1177
3. https://lore.kernel.org/stable/20230227211548.13923-1-kuniyu@amazon.com

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
kernel: bump 5.10 to 5.10.170

No patches modified for this bump

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
kernel: bump 5.10 to 5.10.172

Removed upstreamed:
	backport-5.10/804-0001-net-Remove-WARN_ON_ONCE-sk-sk_forward_alloc-from-sk_.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.171&id=3e4bbd1f38a8d35bd2d3aaffdb5f6ada546b669a

Signed-off-by: John Audia <therealgraysky@proton.me>

hauke commented on target/linux/generic/backport-5.10/811-v6.1-0001-nvmem-core-Fix-memleak-in-nvmem_register.patch:
```diff
@@ -22,10 +22,23 @@ Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
++	if (nvmem->nkeepout) {
++		rval = nvmem_validate_keepouts(nvmem);
++		if (rval)
++			goto err_device_del;
```
```suggestion
+			goto err_put_device;
```
`err_device_del` does not exist here.

I compared drivers/nvmem/core.c between v6.1.15 and 5.10.172 + your patches and got this diff:
````
hauke@hauke-t480:~/linux/linux-next-clean$ git diff HEAD v6.1.15 -- drivers/nvmem/core.c
diff --git a/drivers/nvmem/core.c b/drivers/nvmem/core.c
index 02f4a1ab1b75..34ee9d36ee7b 100644
--- a/drivers/nvmem/core.c
+++ b/drivers/nvmem/core.c
@@ -836,7 +836,7 @@ struct nvmem_device *nvmem_register(const struct nvmem_config *config)
        if (nvmem->nkeepout) {
                rval = nvmem_validate_keepouts(nvmem);
                if (rval)
-                       goto err_device_del;
+                       goto err_put_device;
        }
 
        if (config->compat) {
@@ -1237,16 +1237,21 @@ struct nvmem_cell *of_nvmem_cell_get(struct device_node *np, const char *id)
        if (!cell_np)
                return ERR_PTR(-ENOENT);
 
-       nvmem_np = of_get_next_parent(cell_np);
-       if (!nvmem_np)
+       nvmem_np = of_get_parent(cell_np);
+       if (!nvmem_np) {
+               of_node_put(cell_np);
                return ERR_PTR(-EINVAL);
+       }
 
        nvmem = __nvmem_device_get(nvmem_np, device_match_of_node);
        of_node_put(nvmem_np);
-       if (IS_ERR(nvmem))
+       if (IS_ERR(nvmem)) {
+               of_node_put(cell_np);
                return ERR_CAST(nvmem);
+       }
 
        cell_entry = nvmem_find_cell_entry_by_node(nvmem, cell_np);
+       of_node_put(cell_np);
        if (!cell_entry) {
                __nvmem_device_put(nvmem);
                return ERR_PTR(-ENOENT);
hauke@hauke-t480:~/linux/linux-next-clean$ 
````
If you change `err_put_device` it should be fine.
